test/intl402/DateTimeFormat/prototype/resolvedOptions/hourCycle-default.js

```diff
@@ -7,50 +7,40 @@ description: >
-        ii. Else,
-           1. Assert: hour12 is false.
-           2. If hcDefault is "h11" or "h23", then
-              a. Set hc to "h23".
-           3. Else,
-              a. Set hc to "h24".
-     e. Set dateTimeFormat.[[HourCycle]] to hc.
-     
-locale: [en, fr, it, ja, zh, ko, ar, hi]
+  11.3.7 Intl.DateTimeFormat.prototype.resolvedOptions()
+
+  11.1.2 CreateDateTimeFormat ( dateTimeFormat, locales, options, required, defaults)
+   23. Let dataLocaleData be localeData.[[<dataLocale>]].
+   24. If hour12 is true, then
+       a. Let hc be dataLocaleData.[[hourCycle12]].
+   25. Else if hour12 is false, then
+       a. Let hc be dataLocaleData.[[hourCycle24]].
+   26. Else,
+        a. Assert: hour12 is undefined.
+        b. Let hc be r.[[hc]].
+        c. If hc is null, set hc to dataLocaleData.[[hourCycle]].
+  27. Set dateTimeFormat.[[HourCycle]] to hc.
+
+locale: [en, fr, it, ja, zh, ko, ar, hi, en-u-hc-h24]
 ---*/
 
-let locales = ["en", "fr", "it", "ja", "zh", "ko", "ar", "hi"];
-
-locales.forEach(function(locale) {
-  let hcDefault = (new Intl.DateTimeFormat(locale, {hour: "numeric"}))
-      .resolvedOptions().hourCycle;
-  if (hcDefault == "h11" || hcDefault == "h23") {
-    assert.sameValue("h11",
-        (new Intl.DateTimeFormat(locale, {hour: "numeric", hour12: true}))
-        .resolvedOptions().hourCycle);
-    assert.sameValue("h23",
-        (new Intl.DateTimeFormat(locale, {hour: "numeric", hour12: false}))
-        .resolvedOptions().hourCycle);
-  } else {
-    assert.sameValue(true, hcDefault == "h12" || hcDefault == "h24")
-    assert.sameValue("h12",
-        (new Intl.DateTimeFormat(locale, {hour: "numeric", hour12: true}))
-        .resolvedOptions().hourCycle);
-    assert.sameValue("h24",
-        (new Intl.DateTimeFormat(locale, {hour: "numeric", hour12: false}))
-        .resolvedOptions().hourCycle);
+let locales = ["en", "fr", "it", "ja", "zh", "ko", "ar", "hi", "en-u-hc-h24"];
+
+locales.forEach(function(locale){
+  let hcDefault = new Intl.DateTimeFormat(locale, {hour: "numeric"}).resolvedOptions().hourCycle;
+  if (hcDefault === "h11" || hcDefault === "h12"){
+    assert.sameValue(hcDefault, new Intl.DateTimeFormat(locale, {hour: "numeric", hour12: true}).resolvedOptions().hourCycle);
+
+    // no locale has "h24" as a default. see https://github.com/tc39/ecma402/pull/758#issue-1622377292
+    assert.sameValue("h23", new Intl.DateTimeFormat(locale, {hour: "numeric", hour12: false}).resolvedOptions().hourCycle);
   }
+  
+  // however, "h24" can be set via locale extension.
+  if (hcDefault === "h23" || hcDefault === "h24"){
+    assert.sameValue(hcDefault, new Intl.DateTimeFormat(locale, {hour: "numeric", hour12: false}).resolvedOptions().hourCycle);
+  }
+
+
+  let hcHour12 = new Intl.DateTimeFormat(locale, {hour: "numeric", hour12: true}).resolvedOptions().hourCycle;
+  assert(hcHour12 === "h11" || hcHour12 === "h12", "Expected `hourCycle` to be in [\"h11\", \"h12\"]");
```

Comment:
Please mention the actual value in the error message if you're not using `sameValue`